code/xcvm/cosmwasm/contracts/common/src/gateway.rs

```diff
@@ -21,49 +20,46 @@ pub struct InstantiateMsg {
 	pub admin: String,
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
-#[serde(rename_all = "snake_case")]
+#[cw_serde]
 pub enum ExecuteMsg {
 	IbcSetNetworkChannel {
 		network_id: NetworkId,
 		channel_id: String,
 	},
 
-	ExecuteProgram {
-		/// The program salt.
-		salt: Vec<u8>,
-		/// The program.
-		program: DefaultXCVMProgram,
-		/// Assets to fund the XCVM interpreter instance
-		/// The interpreter is funded prior to execution
-		assets: Funds<Displayed<u128>>,
-	},
+	/// Sent by the user to execute a program on their behalf.
+	ExecuteProgram(ExecuteProgramMsg),
```

Comment:
{
	execute_program: ExecuteProgramMsg,
},

Reply:
IMO that only adds noise to the code.  ExecuteProgramMsg is the entirety of the ExecuteProgram message so it makes sense to use a singleton tuple the same way its used in Bridge below.

Reply:
generally i agree, bot not in this case so

Reply:
Well, what’s different about this case?

Reply:
1. There was case when it was {}, not (). So indicator it can happen again.
2. I have to write EPP { foo : }, I would like to write(or macro) EP { foo : } too. For consistency.
3. If EP will be not { foo: } BUT EP { foo : , tips_fees_time_configs_and_hints_from_end_user : } will not require refactor back. Oh 1 just happened in future.
4. Also refactoring to () was done, it it was not done it would be faster to refactor use {} in first place.

Reply:
Done.